### ISSUE TITLE
"Garbled" character output in PowerShell when launching "git gui"

### ISSUE THREAD

**Issue by patrikhuber** [microsoft/terminal#4921]
# Environment

* OS: Microsoft Windows NT 10.0.18362.0
* PowerShell version: 6.2.4
* posh-git version/path: 1.0.0 beta3 ~\Documents\PowerShell\Modules\posh-git\1.0.0
* git version 2.25.0.windows.1

# Steps to reproduce

In a PowerShell, with posh-git and git-for-windows (which includes git gui) installed:

When typing `git gui`, the git gui window opens, but on the powershell prompt, the encoding of the characters is outputted and the cursor stays on that line. Only after pressing "Enter", the prompt becomes "normal" again.

```
~\MyProj [mybranch +2 ~2 -0 !]> git gui    <-- nicely colourised
~\MyProj [93m[[0m[96mmybranch[0m[31m[49m +2[0m[31m[49m ~2[0m[31m[49m -0[0m[31m ![0m[93m][0m>    <-- no colours, "weird" output. Have to press "enter" to get prompt to normal again.
~\MyProj [mybranch +2 ~2 -0 !]>    <-- nicely colourised
```

![image](https://user-images.githubusercontent.com/4967343/74587540-ab6a4a00-4feb-11ea-93fb-99cadce54d23.png)

# Expected behavior

Quite obvious I think, the PowerShell prompt should output the "normal" prompt with the coloured characters and stuff (from posh-git).

# Actual behavior

Stuff like `[93m[[0m` is output (see above/screenshot).

# Related issues

I've opened https://github.com/dahlbyk/posh-git/issues/738 first but the author suggested it's likely a git-gui issue. I've then opened https://github.com/prati0100/git-gui/issues/34 but the author said it might likely be a Windows Terminal issue and related to https://github.com/microsoft/terminal/issues/4809. After posting in https://github.com/microsoft/terminal/issues/4809, @DHowett-MSFT advised to open a new issue here, as my issue is with PowerShell, and not with git-bash.

**Comment by DHowett-MSFT** [comment]
Alright, I just got this thing under event tracing. Here's what I found (annotated)

![image](https://user-images.githubusercontent.com/14316954/76690446-2e87bb80-65fd-11ea-8f30-f5385a6cfc0f.png)

(time goes from left to right)

What's happening is this:

1. I type in `git gui`, press enter
2. Powershell _disables_ VT rendering (for compatibility reasons), setting mode=3
3. `git.exe` launches, and kicks off a sequence of processes (one of which is `sh.exe`)
4. `git.exe` _exits_, returning control to Powershell
5. Powershell _re-enables_ VT rendering, setting mode=7
6. Powershell prints its prompt, or when it's being slow it _doesn't_ print its prompt.
7. `sh.exe`, which was spawned under line 3, _disables VT rendering_
8. Powershell finishes printing the prompt and begins taking user input
9. When the user types, powershell tries to use VT (disabled on line 7) to color it
10. This happens: ![image](https://user-images.githubusercontent.com/14316954/76690479-8e7e6200-65fd-11ea-8df5-fe7afc0a76dc.png)

Unfortunately, there isn't anything powershell (or the windows console!) can do to help this situation: the input mode is console-wide global state, and sh.exe is setting that global state to ignore powershell's text rendering commands.

**Comment by DHowett-MSFT** [comment]
In light of all this, I've filed /dup #4954 to track figuring out whether we can make the output handle mode process-local state instead of global state.
